include/pressio/solvers_nonlinear/solvers_nonlinear_enums_and_tags.hpp

```diff
@@ -68,6 +68,7 @@ enum class Update
    Armijo,
    LMSchedule1,
    LMSchedule2,
+   BasicBacktrack,
```

Comment:
i think it would be helpful if we could find a better name for this (and yes, a better name for LMSchedule1,2 too). 
If the main feature of this backtracking is that the residual norm is by construction non-increasing, something that expresses that, any suggestion? `Update::NonDecreasingObjective` or something like that

Reply:
i am not going to get nuts about this, so if we don't find something quickly we can leave the name as is

Reply:
There is not a common name that I am aware of. We could do BacktrackNonIncreasingObjective. I would keep Backtrack in the name though as this is what people know these by.

Reply:
Changed this to "BacktrackStrictlyDecreasingObjective"